### ISSUE TITLE
NVDA - ASP.Net Listbox Issue - Unable to switch Focus mode from browse mode

### ISSUE THREAD

**Issue by MohammedAhmed679664** [nvaccess/nvda#11593]
Hi All,

We are using asp.net List box in one of our application and facing below issue in NVDA browse mode.

Issue: When we navigate to List box control and change to Focus mode for items selection, unable to select items by using UP and DOWN arrow keys

Found same issue in below site also.
https://whatsock.com/tsg/Coding%20Arena/ARIA%20Listboxes/Multiselect/demo.htm

Could anyone help us to know is it a NVDA issue with respect to asp.net list box Issue.

Thanks In Advance
Ahmed

**Comment by Adriani90** [comment]
Hello, I am using last NVDA alpha version from
https://www.nvaccess.org/files/nvda/snapshots/
When activating the link in the description and enabling focus mode on the list, I cannot select values. But when I press tab after focus mode is enabled, then the focus moves to the list and I can select values with arrow keys.

For some reason it seems the system focus and the system carret does not synchronize properly with NVDA's virtual cursor on this multi select list.

Is the tabindex set correctly on the list?

Maybe @jcsteh or @JulienCochuyt can also help here.

**Comment by MohammedAhmed679664** [comment]
Hi Team,

Thanks for your response!!

We are using NVDA 2020.2 version and tabindex is set correctly to the list box.
Issue: While navigating browse mode to focus mode, unable to select items of list box.

Verified that this issue is not found with JAWS screen reader. Please let us know if you need any further details.

Regards,
Ahmed

**Comment by feerrenrut** [comment]
Hi @MohammedAhmed679664 Please could you give exact steps to reproduce, ideally fill out our bug report template. The information is very important for helping us understand exactly the behavior you are describing. Right now it is not clear which list on the example page you are talking about. A minimal example would be even better.

The list "What do you want with lunch?" behaved strangely. 
The "Multiselect field"  list behaved as expect.

How this behaves does seem to depends how focus arrives on the object, and how focus mode is triggered in NVDA. 

A simple test case, test cases, and expected / actual behavior would go along way to narrowing the cause of this.

**Comment by MohammedAhmed679664** [comment]
@feerrenrut - please find attached DOM screenshot of List box and we are using ASP.Net List box as below

ASP.Net Code
=========
 **<asp:ListBox role="listbox" runat="server" TabIndex="0" ID="AdministratorList" onKeyDown="keycode(event,this)" onclick="EnableSubmit();" SelectionMode="Multiple" CssClass="listbox"></asp:ListBox>**

In DOM
=====
![ASP net Multi-Select List box](https://user-images.githubusercontent.com/54573286/93095352-9ad49c00-f6c0-11ea-86f7-27af9c3897e3.png)

Issue Description:
===========
I am navigating in the page using NVDA browse mode i.e UP and DOWN arrow keys. As we all know once after reaching to interactive elements, we need to switch to Focus mode(using Insert+SPACEBAR) to operate interactive control. 
In the same way i too once traversing to list box using browse mode, to select list box items in it i switched to Focus mode but i am unable to select list box items.

Hope you have understood the issue. 

Please share the template if you still have any queries in this regard.

**Comment by feerrenrut** [comment]
Sorry, what you have provided is not clear, and does not allow someone to quickly reproduce the issue in the same way as you. Please provide a self contained sample that can be opened directly in the browser, exact steps to reproduce the issue, and the exact text that is spoken as well as the text you think it should say.

The issue template is available when you open a new bug report.

**Comment by Adriani90** [comment]
@feerrenrut here is definitely the multi select list that seems not to work always as expected. Sometimes if you activate focus mode on it and press the arrow keys, NVDA does not say anything. However it should say "apple" ... etc.

**Comment by Adriani90** [comment]
Here is an example of another multi select list, but this seems not supported by NVDA. Not sure what the reason is:
https://asp-net-example.blogspot.com/2013/12/aspnet-listbox-get-multiple-selected.html

**Comment by MohammedAhmed679664** [comment]
Thanks @Adriani90 for giving more details. 

@feerrenrut --What ever @Adriani90 said that is what the issue.

Team, Please confirm us is it a NVDA issue?

**Comment by leonardder** [comment]
I see two list boxes when I tab,navigate through the page referenced in the initial description. IN the first list box, i can arrow and enable/disable entries with space. In the second box, I can arrow through the list and that seems to select things. I assume the first list is meant to be the accessible one and the second list is not.

**Comment by Adriani90** [comment]
This is the test file with the list that reproduces this issue:
https://github.com/nvaccess/nvda/files/5221204/Minimal.test.case.html.txt
